cmd/infra/aws/create.go

```diff
@@ -249,15 +279,131 @@ func (o *CreateInfraOptions) CreateInfra(ctx context.Context, l logr.Logger) (*C
+		// Obtain account IDs for both accounts
+		clusterSTSClient := sts.New(awsSession, awsutil.NewConfig())
+		vpcOwnerSTSClient := sts.New(vpcOwnerAWSSession, awsutil.NewConfig())
+
+		clusterAccountID, err := clusterSTSClient.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
+		if err != nil {
+			return nil, err
+		}
+		vpcOwnerAccountID, err := vpcOwnerSTSClient.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
+		if err != nil {
+			return nil, err
+		}
+		result.VPCCreatorAccountID = aws.StringValue(vpcOwnerAccountID.Account)
+		result.ClusterAccountID = aws.StringValue(clusterAccountID.Account)
+
+		privateSubnetIDsToShare := make([]*string, 0, len(result.Zones))
+		publicSubnetIDsToShare := make([]*string, 0, len(publicSubnetIDs))
+		allSubnetIDsToShare := make([]*string, 0, len(result.Zones)+len(publicSubnetIDs))
+
+		for _, zone := range result.Zones {
+			privateSubnetIDsToShare = append(privateSubnetIDsToShare, aws.String(zone.SubnetID))
+		}
+		for _, subnetID := range publicSubnetIDs {
+			publicSubnetIDsToShare = append(publicSubnetIDsToShare, aws.String(subnetID))
+		}
+
+		allSubnetIDsToShare = append(allSubnetIDsToShare, privateSubnetIDsToShare...)
+		allSubnetIDsToShare = append(allSubnetIDsToShare, publicSubnetIDsToShare...)
+
+		subnetsResult, err := ec2Client.DescribeSubnetsWithContext(ctx, &ec2.DescribeSubnetsInput{
+			SubnetIds: allSubnetIDsToShare,
+		})
+		if err != nil {
+			return nil, err
+		}
+		subnetArns := make([]*string, 0, len(subnetsResult.Subnets))
+		for _, subnet := range subnetsResult.Subnets {
+			subnetArns = append(subnetArns, subnet.SubnetArn)
+		}
+
+		// Share subnets
+		l.Info("Sharing VPC subnets with cluster creator account", "subnetids", allSubnetIDsToShare)
+		ramClient := ram.New(vpcOwnerAWSSession, awsutil.NewConfig())
+		if _, err = ramClient.CreateResourceShareWithContext(ctx, &ram.CreateResourceShareInput{
+			Name:         aws.String(fmt.Sprintf("%s-share", o.InfraID)),
+			Principals:   []*string{aws.String(result.ClusterAccountID)},
+			ResourceArns: subnetArns,
+			Tags: []*ram.Tag{
+				{
+					Key:   aws.String(clusterTag(o.InfraID)),
+					Value: aws.String(clusterTagValue),
+				},
+			},
+		}); err != nil {
+			return nil, err
+		}
+
+		// Wait for subnets to be visible in the cluster creator account
+		backoff := wait.Backoff{
+			Steps:    10,
+			Duration: 30 * time.Second,
+			Factor:   1.0,
+			Jitter:   0.1,
+		}
+		var subnetResult *ec2.DescribeSubnetsOutput
+		if err = retry.OnError(backoff, func(error) bool { return true }, func() error {
+			var err error
+			subnetResult, err = clusterCreatorEC2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: allSubnetIDsToShare,
+			})
+			if err != nil || len(subnetResult.Subnets) != len(allSubnetIDsToShare) {
+				l.Info("Waiting for subnets to be available in cluster creator account")
+				return fmt.Errorf("not ready yet")
+			}
+			return nil
+		}); err != nil {
+			return nil, err
+		}
```

Comment:
Maybe this should be extracted to a separate function for sharing the subnets?

Reply:
Thx, addressed